File: JAVA/hashmap_frequency.java
import java.util.*;
class Solution {
    public HashMap<Integer, Integer> countFrequencies(int[] arr) {
        HashMap<Integer, Integer> freq = new HashMap<>();
        for(int i=0; i<arr.length; i++) {
            int num = arr[i];
            if(freq.containsKey(num)) {
                freq.put(num, freq.get(num) + 1);
            } else {
                freq.put(num, 1);
            }
        }
        return freq;
    }
}
//4 

File: CPP/hashmap_frequency.cpp
#include <vector>
#include <unordered_map>
using namespace std;
unordered_map<int, int> countFrequencies(vector<int>& arr) {
    unordered_map<int,int> freq;
    for(int i=0; i<arr.size(); i++) {
        int num = arr[i];
        if(freq.count(num)) {
            freq[num]++;
        } else {
            freq[num] = 1;
        }
    }
    return freq;
}
